src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -307,16 +300,28 @@ public void deletePaymentMethod(final UUID kbAccountId, final UUID kbPaymentMeth
             } catch (final ServiceException e) {
                 throw new PaymentPluginApiException("Unable to retrieve recurring details in Adyen", e);
             }
-
-            if (!recurringDetailList.isEmpty()) {
-                final String token = recurringDetailList.get(recurringDetailList.size() - 1).getRecurringDetailReference();
+            for (final RecurringDetail recurringDetail : recurringDetailList) {
+                final AdyenResponsesRecord formerResponse;
+                try {
+                    formerResponse = dao.getResponse(recurringDetail.getFirstPspReference());
+                } catch (SQLException e) {
+                    throw new PaymentPluginApiException("Unable to retrieve adyen response", e);
+                }
+                final Payment payment;
+                try {
+                    payment = killbillAPI.getPaymentApi().getPayment(UUID.fromString(formerResponse.getKbPaymentId()), true, properties, context);
```

Comment:
Oh, sure. I had it set to `false` originally; must have switched to `true` to try something and forgot to switch back :innocent: